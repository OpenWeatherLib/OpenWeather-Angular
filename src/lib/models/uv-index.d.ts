export interface UvIndex {
  lat: number;
  lon: number;
  date_iso: Date;
  date: Date;
  value: number;
}
