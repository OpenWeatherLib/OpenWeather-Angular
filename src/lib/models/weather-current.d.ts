import { Clouds } from "@lib/models/clouds";
import { Coordinates } from "@lib/models/coordinates";
import { Main } from "@lib/models/main";
import { Sys } from "@lib/models/sys";
import { WeatherPart } from "@lib/models/weather-part";
import { Wind } from "@lib/models/wind";

export interface WeatherCurrent {
    coord: Coordinates;
    weather: WeatherPart[];
    base: string;
    main: Main;
    visibility: number;
    wind: Wind;
    clouds: Clouds;
    dt: Date;
    sys: Sys;
    id: number;
    name: string;
    cod: number;
}
