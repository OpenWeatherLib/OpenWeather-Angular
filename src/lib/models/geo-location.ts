import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
