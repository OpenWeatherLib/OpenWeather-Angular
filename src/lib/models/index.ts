export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
