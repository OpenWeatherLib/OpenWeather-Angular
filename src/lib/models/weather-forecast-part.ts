import "@lib/extensions/string.extensions";
import { classJsonKey } from "@lib/decorator";
import { ForecastListType } from "@lib/enums";
import WeatherCondition from "@lib/enums/weather-condition.enum";

@classJsonKey(String().empty, "sys")
export class WeatherForecastPart {
    main: string = String().empty;
    weatherCondition: WeatherCondition = WeatherCondition.null;
    description: string = String().empty;
    weatherDefaultIcon: string = String().empty;
    temperature: number = 0.0;
    temperatureMin: number = 0.0;
    temperatureMax: number = 0.0;
    temperatureKf: number = 0.0;
    pressure: number = 0.0;
    pressureSeaLevel: number = 0.0;
    pressureGroundLevel: number = 0.0;
    humidity: number = 0.0;
    cloudsAll: number = 0.0;
    windSpeed: number = 0.0;
    windDegree: number = 0.0;
    dateTime: Date = new Date();
    listType: ForecastListType = ForecastListType.Null;

    isDefault(): boolean {
        return this.main.isNullOrEmpty()
            && this.weatherCondition === WeatherCondition.null
            && this.description.isNullOrEmpty()
            && this.weatherDefaultIcon.isNullOrEmpty()
            && this.temperature === 0.0
            && this.temperatureMin === 0.0
            && this.temperatureMax === 0.0
            && this.temperatureKf === 0.0
            && this.pressure === 0.0
            && this.pressureSeaLevel === 0.0
            && this.pressureGroundLevel === 0.0
            && this.humidity === 0.0
            && this.cloudsAll === 0.0
            && this.windSpeed === 0.0
            && this.windDegree === 0.0
            && this.listType === ForecastListType.Null;
    }
}
