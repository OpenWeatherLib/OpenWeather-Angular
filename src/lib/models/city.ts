import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
