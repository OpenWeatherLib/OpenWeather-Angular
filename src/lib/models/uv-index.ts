import { GeoLocation } from "./geo-location";

export class UvIndex {
  geoLocation: GeoLocation = new GeoLocation();
  date: Date = new Date();
  value: number = 0.0;

  isDefault(): boolean {
    return this.geoLocation.isDefault() && this.value === 0.0;
  }
}
