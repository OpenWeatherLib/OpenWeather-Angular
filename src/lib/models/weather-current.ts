import "@lib/extensions/string.extensions";
import { classJsonKey } from "@lib/decorator";
import WeatherCondition from "@lib/enums/weather-condition.enum";
import { City } from "./city";

@classJsonKey(String().empty, "sys")
export class WeatherCurrent {
    icon: string = String().empty;
    description: string = String().empty;
    weatherCondition: WeatherCondition = WeatherCondition.null;
    temperature: number = 0.0;
    temperatureMin: number = 0.0;
    temperatureMax: number = 0.0;
    humidity: number = 0.0;
    pressure: number = 0.0;
    visibility: number = 0.0;
    cloudsAll: number = 0.0;
    windSpeed: number = 0.0;
    windDegree: number = 0.0;
    dateTime: Date = new Date();
    sunriseTime: Date = new Date();
    sunsetTime: Date = new Date();
    city: City = new City();
    lastUpdate: Date = new Date();

    isDefault(): boolean {
        return this.icon.isNullOrEmpty()
            && this.description.isNullOrEmpty()
            && this.weatherCondition === WeatherCondition.null
            && this.temperature === 0.0
            && this.temperatureMin === 0.0
            && this.temperatureMax === 0.0
            && this.humidity === 0.0
            && this.pressure === 0.0
            && this.visibility === 0.0
            && this.cloudsAll === 0.0
            && this.windSpeed === 0.0
            && this.windDegree === 0.0
            && this.city.isDefault();
    }
}
