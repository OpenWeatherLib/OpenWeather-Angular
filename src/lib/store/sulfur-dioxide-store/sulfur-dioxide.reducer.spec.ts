import { City, SulfurDioxide } from "@lib/models";
import { loadSulfurDioxideRequestAction, loadSulfurDioxideSuccessAction, loadSulfurDioxideErrorAction } from "./sulfur-dioxide.actions";
import { sulfurDioxideReducer } from "./sulfur-dioxide.reducer";
import { SulfurDioxideState } from "./sulfur-dioxide.state";

describe("SulfurDioxide Reducer Tests", () => {

    const initialState: SulfurDioxideState = {
        sulfurDioxide: undefined,
        dateTime: "2019-07-12",
        accuracy: 2,
        isLoading: true,
        error: "Test Error"
    };

    describe("State Changes", () => {
        test("should have initial state after invalid action", () => {
            // Arrange & Act
            const testState = sulfurDioxideReducer(initialState, { type: "INVALID_ACTION" } as any);

            // Assert
            expect(testState).toBe(initialState);
        });

        test("should load city on success of loading", () => {
            // Arrange
            const loadRequest = loadSulfurDioxideRequestAction({ city: {} as City });
            const loadSuccess = loadSulfurDioxideSuccessAction({ sulfurDioxide: {} as SulfurDioxide });

            // Act
            const testState = [loadRequest, loadSuccess].reduce(sulfurDioxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });

        test("should set error on error of loading", () => {
            // Arrange
            const loadRequest = loadSulfurDioxideRequestAction({ city: {} as City });
            const loadError = loadSulfurDioxideErrorAction({ error: "Error" });

            // Act
            const testState = [loadRequest, loadError].reduce(sulfurDioxideReducer, initialState);

            // Assert
            expect(testState).toMatchSnapshot();
        });
    });
});
