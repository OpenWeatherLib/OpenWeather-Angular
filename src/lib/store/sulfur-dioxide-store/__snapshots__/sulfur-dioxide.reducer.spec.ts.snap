// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SulfurDioxide Reducer Tests State Changes should load city on success of loading 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "2019-07-12",
  "error": undefined,
  "isLoading": false,
  "sulfurDioxide": Object {},
}
`;

exports[`SulfurDioxide Reducer Tests State Changes should set error on error of loading 1`] = `
Object {
  "accuracy": 2,
  "dateTime": "2019-07-12",
  "error": "Error",
  "isLoading": false,
  "sulfurDioxide": undefined,
}
`;
